saleor/payment/gateways/razorpay/__init__.py

```diff
@@ -28,32 +26,29 @@
 logger = logging.getLogger(__name__)
 
 
-def _generate_transaction(
-        payment: Payment,
+def _generate_response(
+        payment_information: Dict,
         kind: str,
-        amount: Decimal,
         *,
-        id='',
         is_success=True,
-        **data) -> Transaction:
+        **data) -> Dict:
     """Creates a transaction from a Razorpay's success payload
     or from passed data."""
-    transaction = create_transaction(
-        payment=payment,
-        kind=kind,
-        amount=amount,
-        currency=data.pop('currency', payment.currency),
-        gateway_response=data,
-        token=id,
-        is_success=is_success)
-    return transaction
-
-
-def check_payment_supported(payment: Payment):
-    """Checks that a given payment is supported"""
-    if payment.currency not in SUPPORTED_CURRENCIES:
+    return {
+        'transaction_id': data.get('id', payment_information['token']),
+        'kind': kind,
+        'amount': data.get('amount', payment_information['amount']),
```

Comment:
Do we really need to provide a fallback ```payment_information['amount']``` as ```create_transaction``` has alread provided a default value for ```amount``` and ```currency```? There is a difference between braintree, razorpay and stripe.